### ISSUE TITLE
Replace all doesn't modify any matches (in one of my scripts)

### ISSUE THREAD

**Issue by DanKaplanSES** [Tampermonkey/tampermonkey#1927]
(This template should probably ask for reproducible steps here like this:)

## Reproducible Steps

Before I start, I can only reproduce this issue on this one script. If I create a new one and try the same steps, "replace all" works as expected. Therefore, I think you'll need to use my exact script (see below) in order to reproduce this.

1. Create a new script
2. Copy the script code into it (see below)
3. Save the script
4. Open the script
5. Click Find -> Replace
6. Type "console.log(`" (without double quotes) into the "Replace:" field
7. Press enter
8. Type "console.log(`userscript: " (without double quotes, note the space at the end, though I don't know if that's important) into the "Replace with:" field
9. Press enter
10. This lower menu bar should appear:
   ![image](https://github.com/Tampermonkey/tampermonkey/assets/373675/2ff876d0-c696-4236-9e8d-8d7018375803)
11. Click the "Replace All" button

## Expected Behavior
Running replace all replaces all matches

## Actual Behavior
Running replace all makes no change, but the menu disappears the same way it would if it worked correctly.

## Specifications

- Firefox 120.0.1 (64-bit)
- TM: 5.0.0
- OS: Windows 10

## Script

(Please give an example of the script if applicable.)

```js
// ==UserScript==
// @name         Test. DELETE!!!!!!!!!!!!!!!!!!!!!
// @namespace    http://tampermonkey.net/
// @version      0.1
// @description  This adds a link to the fork's branch on the GitHub "Comparing changes" page. See https://stackoverflow.com/questions/77623282/how-do-i-get-my-remote-branch-url-from-the-github-create-pull-request-page for more details.
// @author       DanKaplanSES
// @match        https://github.com/*/*/compare/*...*
// @icon         https://www.google.com/s2/favicons?sz=64&domain=github.com
// @require      https://code.jquery.com/jquery-3.7.1.min.js
// @grant        none
// ==/UserScript==

console.log(`userscript start`);
jQuery.noConflict(true)(function($) {
    console.log(`window.onload assignment start`);
    window.onload = function () {
        let oldHref = document.location.href;
        const body = document.querySelector("body");
        const observer = new MutationObserver((mutations) => {
            if (oldHref !== document.location.href) {
                console.log(`Modifying page because URL changed from ${oldHref} to ${document.location.href}`);
                oldHref = document.location.href;
                modifyPage();
            }
        });
        observer.observe(body, { childList: true, subtree: true });
    };
    console.log(`window.onload assignment complete`);

    function modifyPage() {
        console.log(`modifyPage() function body start`);
        const browseRepoHtml = `<div class="browse-repo"><a href="#" target="_blank">Browse Repo</a></div>`;
        const browseBranchHtml = `<div class="browse-branch"><a href="#" target="_blank">Browse Branch</a></div>`;
        const browseContainerHtml = `
          <div class="browse-container">
            ${browseRepoHtml}
            ${browseBranchHtml}
          </div>`;

        $(`.range-cross-repo-pair`).append($(browseContainerHtml).css({"display": `flex`}));
        $(`.browse-container > div`).css({"width": `50%`});
        $(`.browse-branch`).css({"text-align": `right`});
        $(`.range-editor .pre-mergability`).css({"vertical-align": `top`});

        $($(`.browse-repo a`)[0]).attr(`href`, `/` + $(`.css-truncate-target`)[0].textContent);
        $($(`.browse-branch a`)[0]).attr(`href`, `/` + $(`.css-truncate-target`)[0].textContent + `/tree/` + $(`.css-truncate-target`)[1].textContent);
        console.log(`modifyPage() function body start`);
    }

    console.log(`initial modifyPage() call start`);
    modifyPage();
    console.log(`initial modifyPage() call complete`);
});
console.log(`userscript complete`);
```

**Comment by derjanb** [comment]
Should be fixed at 5.0.6191 ([crx](https://tampermonkey.net/crx/tampermonkey_beta_current.crx)|xpi in review)

Chrome/Edge users, please download the crx file linked above and drag and drop it to the extensions page `chrome://extensions` (after you've enabled 'Developer Mode').
Firefox users please wait for the review being done and install the [BETA version](https://www.tampermonkey.net/index.php?browser=firefox) or check for BETA version updates at `about:addons` 

For a quick fix please export your settings and scripts as zip or (JSON) file at the "Utilities" tab and import it back at the fixed BETA version.

**Comment by DanKaplanSES** [comment]
Thanks. @derjanb should I close this issue?
